pkg/runtime/framework/plugins/torch/torch.go

```diff
@@ -245,18 +269,112 @@ func calculateNumProcPerNode(
+		} else {
+			recipe = constants.TorchTuneFullFinetuneSingleDevice
+			suffix = constants.TorchTuneFullFinetuneSingleDeviceConfigSuffix
+		}
+	} else if numNodes == 1 && isUseQLoraFinetune(trainJob.Spec.Trainer.Args) {
+		recipe = constants.TorchTuneLoRAFinetuneDistributed
+		suffix = constants.TorchTuneQLoRAFinetuneDistributedConfigSuffix
+	} else if numNodes == 1 && isLoraConfigEnabled(trainJob.Spec.Trainer.Args) {
+		recipe = constants.TorchTuneLoRAFinetuneDistributed
+		suffix = constants.TorchTuneLoRAFinetuneDistributedConfigSuffix
 	} else if numNodes > 1 {
 		suffix = constants.TorchTuneFullFinetuneMultiNodesConfigSuffix
 	}
 
-	return recipe, fmt.Sprintf("%s%s", model, suffix)
+	return recipe, fmt.Sprintf("%s%s", getModelFromRuntimeRef(trainJob.Spec.RuntimeRef.Name), suffix)
+}
+
+// getGPUCountPerNode returns the GPU count and a boolean indicating whether a GPU resource was found.
+func getGPUCountPerNode(resourcePerNode corev1.ResourceList) (int, bool) {
+	for resName, resQ := range resourcePerNode {
+		if strings.Contains(strings.ToLower(resName.String()), "gpu") {
+			return int(resQ.Value()), true
+		}
+	}
+	return 0, false
+}
+
+// isLoraConfigEnabled checks if we enables LoraConfig.
+func isLoraConfigEnabled(args []string) bool {
+	for _, arg := range args {
+		if strings.Contains(arg, constants.TorchTuneLoraAttnModules) {
+			return true
+		}
+	}
+	return false
+}
+
+// isUseQLoraFinetune checks if QLoRA fine-tuning should be used.
+func isUseQLoraFinetune(args []string) bool {
+	hasQuantizeBase := false
+	for _, arg := range args {
+		switch {
+		case strings.Contains(arg, constants.TorchTuneUseDora):
+			// If Dora is enabled, no need to continue
+			return false
+		case strings.Contains(arg, constants.TorchTuneQuantizeBase):
+			hasQuantizeBase = true
+		}
+	}
+	return hasQuantizeBase
+}
+
+// extractResourcePerNodeFromRuntime extracts the resource per node from the TorchTune Trainer Node.
+func extractResourcePerNodeFromRuntime(info *runtime.Info) *corev1.ResourceRequirements {
+	if jobSetSpec, ok := runtime.TemplateSpecApply[jobsetv1alpha2ac.JobSetSpecApplyConfiguration](info); ok {
+		for _, rJob := range jobSetSpec.ReplicatedJobs {
+			if rJob.Name != nil && *rJob.Name == constants.Node || rJob.Template.Labels[constants.LabelTrainJobAncestor] == constants.AncestorTrainer {
+				for _, container := range rJob.Template.Spec.Template.Spec.Containers {
+					if container.Name != nil && *container.Name == constants.Node && container.Resources != nil {
+						defaultLimitResource, defaultRequestResource := corev1.ResourceList{}, corev1.ResourceList{}
+						if container.Resources.Limits != nil {
+							defaultLimitResource = *container.Resources.Limits
+						}
+						if container.Resources.Requests != nil {
+							defaultRequestResource = *container.Resources.Requests
+						}
+						return &corev1.ResourceRequirements{
+							Limits:   defaultLimitResource,
+							Requests: defaultRequestResource,
+						}
```

Comment:
This can be simplified as follows:
```go
	res := &corev1.ResourceRequirements{
		Limits:   corev1.ResourceList{},
		Requests: corev1.ResourceList{},
	}
	if container.Resources.Limits != nil {
		res.Limits = *container.Resources.Limits
	}
	if container.Resources.Requests != nil {
		res.Requests = *container.Resources.Requests
	}
	return res
```